### ISSUE TITLE
Support ACM in teleport-cluster helm chart

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#13223]
What would you like Teleport to do?

Currently, the helm chart exposes a way to annotate the service object.

```
       service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:us-east-2:024167042532:certificate/ff56bd55-ebe9-4384-add0-ea87b4923347
       service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
```

This does allow a possibility for setting up an ACM cert, but there are several issues that prevents it from fully working.

* By default, eks will create a classic load balancer. Its TLS client doesn't support the ciphers that teleport uses for TLS 1.2. Connections to the load balancer end up with teleport logging the following:

```
2022-06-06T20:52:33Z WARN [ALPN:PROX] Failed to handle client connection. error:[
ERROR REPORT:
Original Error: *errors.errorString tls: no cipher suite supported by both client and server
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:376 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
	/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:314 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
	/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: tls: no cipher suite supported by both client and server] alpnproxy/proxy.go:322
```

Switching to an nlb be setting the `service.beta.kubernetes.io/aws-load-balancer-type: nlb` annotation does get past that initial issue.

The problem is that all the other ports are set up as http target groups, while port 443 is correctly set as an https target group.

In an attempt to make all ports but 443 have a tcp target mode, I tried setting `service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp`. Unfortunately, this made _all_ the target groups be in TCP mode, including the one for port 443.

I came across [this kubernetes/kubernetes issue](https://github.com/kubernetes/kubernetes/issues/40244), which pointed to the `appProtocol` field as the path forward. I was not able to get this working in my local setup, but it may be a matter of having a newer version of the aws load balancer controller. If appProtocol is indeed all that needs to be added to get this working, then adding tcp or https to each port in https://github.com/gravitational/teleport/blob/6a9b770fb9321d30676af3a2eb7322965d176791/examples/chart/teleport-cluster/templates/service.yaml#L25-L59

If that does not do the trick, then it might be necessary to split the services up into two load balancers-- one for port 443 and one for all the other ports in tcp mode. This would likely need a second DNS name parameter to be used throughout the templates.

What problem does this solve?

acm cert on load balancer not working with teleport-cluster helm chart.

If a workaround exists, please include it.

manually adjust the nlb configuration after kubernetes automation creates it (not a real workaround since it isn't production-appropriate)

gz#5432

**Comment by r0mant** [comment]
@programmerq AWS load balancer controller has `service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy` annotation that allows to tweak the SSL security policy. Have you tried using it?

**Comment by webvictim** [comment]
Also, the annotations needed are clearly stated in the docs: https://goteleport.com/docs/kubernetes-access/helm/guides/aws/#step-4b-configure-teleport-to-use-acm-to-handle-tls

Maybe this needs an explicit callout to remind the user to also use `chartMode: aws` as stated further up in the guide?

**Comment by programmerq** [comment]
I found that I was able to get everything working correctly with the following annotations appearing on my final kube svc object:

```
service.beta.kubernetes.io/aws-load-balancer-backend-protocol: ssl
service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: true
service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:us-east-2:...:certificate/.......
service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
service.beta.kubernetes.io/aws-load-balancer-type: nlb
```

For anyone else who runs into the 'no cipher suite supported by both client and server' error, make sure that you are using an nlb instead of a classic load balancer. Port 443 should have the acm certificate associated with it, and the target group should be the teleport web port in tls mode.

NLB listeners in my working environment (as set up by kube with the above annotations):

![image](https://user-images.githubusercontent.com/37278/172676661-2e8e314d-1937-405f-85a4-8c25eed64a92.png)

the target group for the web port in TLS mode:

![image](https://user-images.githubusercontent.com/37278/172677020-274cd7f4-5595-4ffb-85a4-11d6ab70781d.png)

the target group for the teleport ssh proxy port in TCP mode:

![image](https://user-images.githubusercontent.com/37278/172677133-84ecabcc-631c-4d60-a4fe-48338ab6c672.png)

All other ports should be in TCP mode like the ssh proxy port target group shows.

Closing this one out since it was my error-- I tried several combinations of aws-load-balancer-type, aws-load-balancer-backend-protocol, and aws-load-balancer-ssl-ports but hadn't tried the correct configuration noted in this comment.

Thanks to @webvictim for getting eyes on this request!
